llvm/include/llvm/BinaryFormat/ELF.h

```diff
@@ -2018,6 +2019,9 @@ uint16_t convertArchNameToEMachine(StringRef Arch);
 /// Convert an ELF's e_machine value into an architecture name.
 StringRef convertEMachineToArchName(uint16_t EMachine);
 
+// Convert a triple's architecture to ELF's e_machine value.
+uint16_t convertTripleArchTypeToEMachine(Triple::ArchType ArchType);
```

Comment:
I wonder if there is a good place to share this - feels like something we would need in the MC layer for the object file emission.

Reply:
There is `ELFObjectFile::getArch()` going in the other direction, with pretty minimal coverage.
AFAICT the actual emission is set by each `MCELFObjectTargetWriter` subclass's constructor.
I don't really see any straightforward way to share a single source of truth there.